File: voting-service/src/main/java/com/akerke/votingservice/vote/VoteListener.java
package com.akerke.votingservice.vote;

import com.akerke.votingservice.vote.dto.VoteDTO;
import lombok.extern.slf4j.Slf4j;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Service;

@Service
@Slf4j
public class VoteListener {


    @KafkaListener(topics = "votes", groupId = "vote_group")
    public void listen(VoteDTO voteDTO) {
        log.info("Received vote: " + voteDTO);
    }
}

File: notification-service/src/main/kotlin/com/akerke/notificationservice/notification/VoteListener.kt
package com.akerke.notificationservice.notification

import com.akerke.notificationservice.notification.dto.VoteDTO
import lombok.extern.slf4j.Slf4j
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.stereotype.Component

@Component
@Slf4j
class VoteListener (
        private val notificationService: NotificationService
) {

    @KafkaListener(topics = ["votes"], groupId = "vote_group")
    fun listen(voteDTO: VoteDTO) {
        notificationService.checkForNotifications(voteDTO);
    }

}